framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -258,14 +277,53 @@ IterationAdaptiveDT::computeFailedDT()
   return _dt * _cutback_factor;
 }
 
+bool
+IterationAdaptiveDT::converged()
+{
+  if (!_reject_large_step)
+    return TimeStepper::converged();
+
+  // the solver has not converged
+  if (!TimeStepper::converged())
+    return false;
+
+  // we are already at dt_min or at the start of the simulation
+  // in which case we can move on to the next step
+  if (_dt == _dt_min || _t_step < 2)
+    return true;
+
+  // we need to update the post-processor value
+  // this is the only way I could find to do it, maybe there is a cleaner way
+  // (otherwise, _pps_value is the value computed at the previous time step)
+  if (_pps_value)
+    _fe_problem.execute(EXEC_TIMESTEP_END);
+
+  // we get what the next time step should be
+  Real dt_test = _dt;
+  limitDT(dt_test);
+
+  // we cannot constrain the time step any further
+  if (dt_test == 0)
+    return true;
+
+  // if the time step is much smaller than the current time step
+  // we need to repeat the current iteration with a smaller time step
+  // (the 0.5 value needs to be defined by the user)
```

Comment:
This comment about 0.5 seems to be outdated, presumably this is the `_large_step_rejection_threshold` parameter now.